Replication Project_Final.Rmd

```diff
@@ -0,0 +1,858 @@
+            subs_units = sum(subs_units), total_units = sum(total_units), pct.subs = (subs_units/total_units) * 100,
+            corn.tons = sum(corn.tons), corn.ha = sum(corn.ha), log.corn.yield = log10(corn.tons/corn.ha),
+            total.coffee.units = sum(total.coffee.units), pct.coffee = (total.coffee.units/total_prod_units) * 100, 
+            plot.total = sum(plot.total), units.to5 = sum(units.to5), pct.upto5 = (units.to5/plot.total) * 100, area.total = sum(area.total), log.avg.plot = log10(area.total/plot.total)) %>%
+  select(-no.literacy, -indi, -pop.less.2500, -doctors, -fem.house, -total.house, -young.total, -sqkm, -hom.total, -ejidal, -total_area, -comunal, -private, -total_prod_units, -corn.tons, -total.cattle.units, -subs_units, -total_units, -corn.tons, -corn.ha) %>%
+  mutate(muncode = distrito + 20000)
+
+oaxaca.dist$pct.coffee[is.na(oaxaca.dist$pct.coffee)] <- 0
+
+## ADDING SD ELEVATION WEIGHTED BY POPULATION TO OAXACA
+
+## join population and elevation data frame with distritos in oaxaca 
+pop_elev.dis <- left_join(pop_elev.oax, distritos, by = "muncode")
+
+## calculate sd elevation weighted by population grouped by distrito
+weigh_elev.oax <- pop_elev.dis %>% 
+  group_by(distrito) %>% 
+  summarise(weigh.elev = wt.sd(elev, total.pop)) %>% # use SDMTools to calculate weighted sd
+  mutate(muncode = 20000 + distrito) %>%
+  select(-distrito)
+
+## join weighted elevation with oaxaca main data frame
+oaxaca.dist <- left_join(oaxaca.dist, weigh_elev.oax, by = "muncode")
+
+#calculate theil index for oaxaca by distrito. join distritos with total units for oaxaca, calculate values for distritos
+total_units.oax <- left_join(distritos, total_units.oax, by = "muncode") %>%
+  group_by(distrito) %>%
+  summarise(plot.total = sum(plot.total), area.total = sum(area.total))
+
+#join distritos and plot dized for oaxaca, calculate units and area by size for distritos
+plot_size.oax <- left_join(distritos, plot_size.oax, by = "muncode") %>%
+  select(muncode, distrito, units, area.ha, size) %>%
+  filter(!is.na(size)) %>%
+  group_by(distrito, size) %>%
+  summarise(units = sum(units, na.rm = TRUE), area.ha = sum(area.ha, na.rm = TRUE))
+
+#join plot size and totals for distritos in oaxaca in order to calculate theil index
+theil_prep.oax <- left_join(plot_size.oax, total_units.oax, by = "distrito")
+
+#calculate theil index and turn distritos into muncodes. Not sure about what to do with Inf and NaN values, removed NAs from calculation in order to keep more observations
+theil_index.oax <- theil_prep.oax %>%
+  mutate(z1 = (area.ha/units)/(area.total/plot.total)) %>% # calculate z1
+  group_by(distrito, plot.total) %>%
+  summarise(summation = sum((units*z1)*log(z1), na.rm = TRUE)) %>% #calculate summation
+  mutate(theil = (1/plot.total)*summation) %>%
+  select(distrito, theil) %>%
+  mutate(muncode = distrito + 20000)
+  
```

Comment:
this is the second place that you calculate the Theil index.  You might want to consider writing a function to do this.